packages/cdk8s-plus-17/src/statefulset.ts

```diff
@@ -0,0 +1,217 @@
+import { ApiObject, ApiObjectMetadataDefinition, Lazy, Names } from 'cdk8s';
+import { Construct } from 'constructs';
+import { Resource, ResourceProps } from './base';
+import { Container, ContainerProps } from './container';
+import * as k8s from './imports/k8s';
+import { RestartPolicy, PodTemplate, IPodTemplate, PodTemplateProps } from './pod';
+import { Service, ServiceProps, ServiceType } from './service';
+import { IServiceAccount } from './service-account';
+import { Volume } from './volume';
+
+
+export enum PodManagementPolicy {
+  ORDERED_READY = 'OrderedReady',
+  PARALLEL = 'Parallel',
+}
+
+/**
+ * Properties for initialization of `StatefulSet`.
+ */
+export interface StatefulSetProps extends ResourceProps, PodTemplateProps {
+  /**
+    * Number of desired pods.
+    *
+    * @default 1
+    */
+  readonly replicas?: number;
+
+  /**
+    * Automatically allocates a pod selector for this statefulset.
+    *
+    * If this is set to `false` you must define your selector through
+    * `statefulset.podMetadata.addLabel()` and `statefulset.selectByLabel()`.
+    *
+    * @default true
+    */
+  readonly defaultSelector?: boolean;
+
+  /**
+    * Pod management policy to use for this statefulset.
+    *
+    * @default PodManagementPolicy.ORDERED_READY
+    */
+  readonly podManagementPolicy?: PodManagementPolicy;
+
+  /**
+    * Details of service to create and expose.
+    *
+    * @default undefined - allocate a default service based upon servicePort and serviceName.
+    */
+  readonly service?: ServiceProps;
+
+  /**
+    * Default port to expose if service doesn't includes ports.
+    *
+    * Must set this or include ports in service props.
+    *
+    * @default undefined
+    */
+  readonly servicePort?: number;
+
+  /**
+    * Name to use for the service to allow finding it in DNS.
+    *
+    * @default undefined
+    */
+  readonly serviceName?: string;
```

Comment:
same

Reply:
We would still need it if the Service was optional.  But if the service was required we could remove it yes.